src/uct/rocm/Makefile.am

```diff
@@ -12,7 +12,10 @@ libuct_rocm_la_CPPFLAGS = $(BASE_CPPFLAGS) $(ROCM_CPPFLAGS)
 libuct_rocm_la_CFLAGS   = $(BASE_CFLAGS)
 libuct_rocm_la_LIBADD   = $(top_builddir)/src/ucs/libucs.la \
                           $(top_builddir)/src/uct/libuct.la
-libuct_rocm_la_LDFLAGS  = $(ROCM_LDFLAGS) $(ROCM_LIBS) -version-info $(SOVERSION)
+libuct_rocm_la_LDFLAGS  = $(ROCM_LDFLAGS) $(ROCM_LIBS) -version-info $(SOVERSION) \
+                          $(patsubst %, -Xlinker %, -L$(ROCM_ROOT)/lib -rpath $(ROCM_ROOT)/hip/lib -rpath $(ROCM_ROOT)/lib) \
+                          $(patsubst %, -Xlinker %, --enable-new-dtags) \
+                          $(patsubst %, -Xlinker %, -rpath $(ROCM_ROOT)/lib64)
```

Comment:
seems like du(tri)plication, can we move these to common var which is set in config/m4/rocm.m4?

Reply:
seems like, yes. But putting them in rocm.m4 kept failing with errors, m4 parse errors.
Expedited these changes as these are needed for correct builds of UCX for rocm IMHO.
Any ideas on how to consolidate them in rocm.m4 would be good to know.

Reply:
wierd.. did you try to just add them to ROCM_LIBS? without the patsubst thing, just prefixing each flag with Xlinker?

Reply:
I tried different things, don't have a log of them. Then, followed the original coding which had ROCM_LDFLAGS/HIP_LDFLAGS in Makefile.am of individual modules. I don't think that was duplication as they are separate modules, and could be revised independently. So, instead of overengineering for no perceivable value add, just followed the existing pattern.  I don't that that should be an issue to move this PR forward.
Future refactoring is not precluded by this PR.

Reply:
@srinivamd it doesn't seem very clean to me, but of someone else from AMD can peer-approve it then it's ok.
BTW do we need this PR in master branch as well?

Reply:
@yosefe I have added @souravzzz for review, not sure whether there are others. 
I understand your point about "clean" but I think these changes would be needed for correctness as multiple versions of ROCm could be installed on a system (explained in my PR comments).

These changes would also be needed in master branch as well.

Until integration, I think, maybe folks can pull in these changes as a patch on to their local workarea.